json_schemas/schema_validator.py

```diff
@@ -0,0 +1,121 @@
+      submitty_validate_schema(testcase, submission_limit_schema, None, None, descriptive_title='', required=False)
+      testcase_num += 1
+      continue
+    except: 
+      pass
+
+    #validate all of the different chunks of the testcase
+    submitty_validate_schema(testcase, testcase_properties_schema, 'dispatcher_actions', 'dispatcher_actions', descriptive_title='testcase {0} dispatcher actions'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'actions', 'actions', descriptive_title='testcase {0} graphics actions'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'points', 'points', descriptive_title='testcase {0} points'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'type', 'type', descriptive_title='testcase {0} testcase type'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'pre_commands', 'pre_commands', descriptive_title='testcase {0} pre-commands'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'single_port_per_container', 'single_port_per_container', descriptive_title='testcase {0} single_port_per_container option'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'use_router', 'use_router', descriptive_title='testcase {0} use_router option'.format(testcase_num))
+    submitty_validate_schema(testcase, testcase_properties_schema, 'title', 'title', descriptive_title='testcase {0} title'.format(testcase_num))
+
+    containers = testcase['containers']
+    container_num = 0
+    #validate each container in the testcase
+    for container in containers:
+      submitty_validate_schema(container, container_schema, None, None, descriptive_title='testcase {0} container {1}'.format(testcase_num, container_num))
+      container_num +=1
+
+    validators = testcase['validation']
+    validator_num = 0
+    #validate each validation object in the testcase
+    for validator in validators:
+      submitty_validate_schema(validator, abstract_validation_object_schema, None, None, descriptive_title='testcase {0} validator {1} (generic)'.format(testcase_num, validator_num))
+      submitty_validate_schema(validator, specific_validation_object_schema, None, None, descriptive_title='testcase {0} validator {1} (specific)'.format(testcase_num, validator_num))
+      validator_num += 1
+
+    testcase_num+=1
+  #Finally, validate the config as a whole.
+  submitty_validate_schema(config_json, schema, None, None, descriptive_title='whole schema')
+
+#We validate a key within a config and a schema. If no key is given, we evaluate the whole config or schema.
+def submitty_validate_schema(config_json, schema, config_json_key, schema_key, descriptive_title='', required=True):
```

Comment:
I'd add a default value of `None` for `config_json_key` and `schema_key` here. Then above, anytime you would have passed in `None` to the function, you can just drop the arguments (e.g. on line 68 and 69).